v2/blockstore/readonly.go

```diff
@@ -331,6 +347,10 @@ func (b *ReadOnly) AllKeysChan(ctx context.Context) (<-chan cid.Cid, error) {
 	// We release the lock when the channel-sending goroutine stops.
 	b.mu.RLock()
 
+	if b.closed {
+		return nil, errClosed
+	}
```

Comment:
What would happen if blockstore is closed while iterating through keys? Do we want to return consistent error there too ?

Reply:
We use mutexes, so a Close/Discard call will block if an AllKeysChan operation is happening. The docs already warn about this.

Reply:
OK that makes sense; I see `closeWithoutMutex` grabs the lock both in RW and RO; all good 👍